File: fixtures/scale/ruby/large/lib/check_bitfold.rb
require_relative 'bitfold_encrypt'
require_relative 'bitfold_decrypt'
require_relative 'corpus_mask'

def check_bitfold
  corpus_mask.each do |text|
    sealed = bitfold_encrypt(text, 3)
    opened = bitfold_decrypt(sealed, 3)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_bitfold.py
from cipherlab.ciphers.bitfold import bitfold_encrypt, bitfold_decrypt
from cipherlab.specs.bitfold_spec import bitfold_spec
from cipherlab.corpus.corpus_mask import corpus_mask


def check_bitfold():
    spec = bitfold_spec()
    for text in corpus_mask():
        sealed = bitfold_encrypt(text, spec["key"])
        opened = bitfold_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "mask"
